### PR TITLE
Update trait model

### PR DESCRIPTION
**Pull request by daikitag**
Update trait model to incorporate multiple distributions.

### CODE REVIEW

**Comment by daikitag** [comment]
unit test failing, as the unit tests are in the previous versions

File: tstrait/trait_model.py [hunk 1]
```diff
@@ -9,139 +9,277 @@ class TraitModel:
...
-    def sim_effect_size(self, num_causal, allele_freq, rng):
+    def _check_parameter(self, num_causal, rng):
+        if not isinstance(num_causal, numbers.Number):
+            raise TypeError("Number of causal sites must be an integer")
+        if int(num_causal) != num_causal or num_causal <= 0:
+            raise ValueError("Number of causal sites must be a positive integer")
+        if not isinstance(rng, np.random.Generator):
+            raise TypeError("rng must be a numpy random generator")
+
+    def frequency_dependence(self, allele_freq, alpha):
```
Comment (jeromekelleher) [inline tstrait/trait_model.py:25]: I don't think we're using this function here any more, so it can be removed?

**Review by jeromekelleher — COMMENTED** [review]
Looks like most of the failures are like this:
```
>       model = trait_model.TraitModelAlleleFrequency(0, 1, -1)
764
E       AttributeError: 'function' object has no attribute 'TraitModelAlleleFrequency'
```
which should be an easy fix?

**Comment by daikitag** [comment]
> Looks like most of the failures are like this:
> 
> ```
> >       model = trait_model.TraitModelAlleleFrequency(0, 1, -1)
> 764
> E       AttributeError: 'function' object has no attribute 'TraitModelAlleleFrequency'
> ```
> 
> which should be an easy fix?

Thank you for your feedback. I modified the unit tests and the functions to simulate quantitaive traits and effect sizes, and shall I create a new branch and a new pull request to show you those changes?

File: tstrait/trait_model.py (hunk 1 above)
Comment (daikitag) [reply tstrait/trait_model.py]: I just modified it. Thank you for your suggestion.

**Comment by daikitag** [comment]
@jeromekelleher 
I modified the `sim_phenotype.py` function and the pytests, such that they work for the current version.

**Comment by daikitag** [comment]
@jeromekelleher 
Thank you very much for your prompt feedback. I have finished doing all the steps that you have mentioned, and would it be possible for you to see it whenever you have some time? This is going to be my first pull request that will be merged to the github repo, so I'm excited about it.

**Review by jeromekelleher — APPROVED** [review]
Looks great :tada:

**Pull request merged**